### ISSUE TITLE
Mismatching model on the MS2Deepscore V2.0 Zenodo repository?

### ISSUE THREAD

**Issue by AharoniLab** [matchms/ms2deepscore#218]
Hi,

Thanks for inspiring work.

Following the basic instructions for using a pretrained model to compute spectral similarities (https://github.com/matchms/ms2deepscore/tree/main) and loading the 'ms2deepscore_model.pt' model from here: https://zenodo.org/records/10814307

...I get the following warning:

> The model version (0.5.0) does not match the version of MS2Deepscore (2.0.0), consider downloading a new model or changing the MS2Deepscore version

Could you kindly clarify?

Also, a slightly less generic file name might be helpful

Thanks,-)

**Comment by niekdejonge** [comment]
Hi @AharoniLab, 
This was a mistake on our end, thanks for notifying us! We forgot updating the version of MS2Deepscore before training that model, which results in this warning.
However, this model should actually work with version 2.0.0 (and not 0.5.0)
So, you can safely use the model and we will fix the warning for future releases of the model.

**Comment by AharoniLab** [comment]
Hi Niek,
Thanks a lot for updating.  The model indeed seems to be able to handle mixed-mode data, to a certain extent (more on that in another thread). I do however, get an error message when trying to setup Monte Carlo predictions (i.e. inference) when using the same mentioned model:

```
MM_similarity_measure = MS2DeepScoreMonteCarlo(model, n_ensembles=10)
_Monte Carlo Dropout is not supposed to be used with a model where dropout-rate=0_
```

That model has indeed a dropout rate of zero, but was it even designed to be used with this certainty estimation approach?

Thanks, Nir.

**Comment by niekdejonge** [comment]
Hi @AharoniLab Sorry I missed this question before. 

This new model is indeed not designed to be used with MonteCarlo dropout. In principle a new one could be trained to make monte carlo dropout work again for the newer versions (with dropout enabled), but we did run into some issues here. So instead, we are working on a new way of assessing prediction certainty as an alternative to monte carlo dropout. We are still in the process of further testing this new certainty estimation method, but will release clearer tutorials on how to do this as soon as we are fully confident that it works well. 

So for now it is not yet easy to use our 2.0 model with monte carlo dropout. 

Another note. The model you are using above had a bug in the training method resulting in suboptimal performance, so a new model has been released on the same zenodo link that has this fixed, resulting in quite a bit better performance. https://zenodo.org/records/13897744
